frame/uniswap-v2/src/tests.rs

```diff
@@ -3,7 +3,39 @@ use frame_support::assert_ok;
 use crate::mock::*;
 use composable_traits::dex::CurveAmm as ConstantProductAmmTrait;
 use frame_support::traits::fungibles::{Inspect, Mutate};
-use sp_runtime::Permill;
+use proptest::prelude::*;
+use sp_runtime::{helpers_128bit::multiply_by_rational, Permill};
+
+/// Accepts -2, -1, 0, 1, 2
+macro_rules! prop_assert_zero_epsilon {
+	($x:expr) => {{
+		let epsilon = 2;
+		let upper = 0 + epsilon;
+		let lower = 0;
+		prop_assert!(upper >= $x && $x >= lower, "{} => {} >= {}", upper, $x, lower);
+	}};
+}
+
+/// Accept a 'dust' deviation
+macro_rules! prop_assert_epsilon {
+	($x:expr, $y:expr) => {{
+		let precision = 1000;
+		let epsilon = 5;
+		let upper = precision + epsilon;
+		let lower = precision - epsilon;
+		let q = multiply_by_rational($x, precision, $y).expect("qed;");
+		prop_assert!(
+			upper >= q && q >= lower,
+			"({}) => {} >= {} * {} / {} >= {}",
+			q,
+			upper,
+			$x,
+			precision,
+			$y,
+			lower
+		);
+	}};
+}
```

Comment:
Some functions have been factored out in `composable-test-helpers` https://github.com/ComposableFi/composable/tree/main/frame/composable-tests-helpers.

Please use this lib so that they are reusable :)